clients/pkg/logentry/stages/extensions_test.go

```diff
@@ -88,49 +89,77 @@ var (
+	labels model.LabelSet
+	line   string
+}
+
 func TestCRI_tags(t *testing.T) {
 	cases := []struct {
 		name            string
 		lines           []string
 		expected        []string
 		maxPartialLines int
+		entries         []testEntry
 		err             error
 	}{
 		{
 			name: "tag F",
-			lines: []string{
-				"2019-05-07T18:57:50.904275087+00:00 stdout F some full line",
-				"2019-05-07T18:57:55.904275087+00:00 stdout F log",
+			entries: []testEntry{
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout F some full line", labels: model.LabelSet{"foo": "bar"}},
+				{line: "2019-05-07T18:57:55.904275087+00:00 stdout F log", labels: model.LabelSet{"foo": "bar"}},
 			},
 			expected: []string{"some full line", "log"},
 		},
 		{
-			name: "tag P",
-			lines: []string{
-				"2019-05-07T18:57:50.904275087+00:00 stdout P partial line 1 ",
-				"2019-05-07T18:57:50.904275087+00:00 stdout P partial line 2 ",
-				"2019-05-07T18:57:55.904275087+00:00 stdout F log finished",
-				"2019-05-07T18:57:55.904275087+00:00 stdout F another full log",
+			name: "tag P multi-stream",
+			entries: []testEntry{
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout P partial line 1 ", labels: model.LabelSet{"foo": "bar"}},
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout P partial line 2 ", labels: model.LabelSet{"foo": "bar2"}},
+				{line: "2019-05-07T18:57:55.904275087+00:00 stdout F log finished", labels: model.LabelSet{"foo": "bar"}},
+				{line: "2019-05-07T18:57:55.904275087+00:00 stdout F another full log", labels: model.LabelSet{"foo": "bar2"}},
 			},
 			expected: []string{
-				"partial line 1 partial line 2 log finished",
+				"partial line 1 log finished",     // belongs to stream `{foo="bar"}`
+				"partial line 2 another full log", // belongs to stream `{foo="bar2"}
+			},
+		},
+		{
+			name: "tag P multi-stream with maxPartialLines exceeded",
+			entries: []testEntry{
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout P partial line 1 ", labels: model.LabelSet{"label1": "val1", "label2": "val2"}},
+
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout P partial line 2 ", labels: model.LabelSet{"label1": "val1"}},
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout P partial line 3 ", labels: model.LabelSet{"label1": "val1", "label2": "val2"}},
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout P partial line 4 ", labels: model.LabelSet{"label1": "val3"}},
+				{line: "2019-05-07T18:57:50.904275087+00:00 stdout P partial line 5 ", labels: model.LabelSet{"label1": "val4"}}, // exceeded maxPartialLines as already 3 streams in flight.
+				{line: "2019-05-07T18:57:55.904275087+00:00 stdout F log finished", labels: model.LabelSet{"label1": "val1", "label2": "val2"}},
+				{line: "2019-05-07T18:57:55.904275087+00:00 stdout F another full log", labels: model.LabelSet{"label1": "val3"}},
+				{line: "2019-05-07T18:57:55.904275087+00:00 stdout F another full log", labels: model.LabelSet{"label1": "val4"}},
```

Comment:
Maybe have a different text here so you can distinguish between this and the previous line.

Reply:
the labelset is different for both lines.

Reply:
Yes, but you don't see that in the expected output. Anyway, it's not a blocker.

Reply:
Actually it does :)

Line 146 and Line 147 (merged with other line)

Reply:
Yeah, but you're not sure if there is not a bug that accidentally merged `another full log` from `val3` instead of `val4`. Unlikely and there are probably other tests to check this of course.

Reply:
Thanks for the using the "Yet Another ... " meme :).